crates/nu-parser/src/parser.rs

```diff
@@ -2915,6 +2915,8 @@ pub fn parse_var_with_opt_type(
         || bytes.contains(&b'"')
         || bytes.contains(&b'\'')
         || bytes.contains(&b'`')
+        || String::from_utf8_lossy(&bytes).parse::<f64>().is_ok()
+        || String::from_utf8_lossy(&bytes).parse::<i64>().is_ok()
```

Comment:
Can you just check if the first byte is a number? That'd be a little cheaper and prevent a similar case of errors